source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/EntityConfigurations/ChargeEntityConfiguration.cs

```diff
@@ -28,24 +28,41 @@ public void Configure(EntityTypeBuilder<Charge> builder)
             builder.ToTable(_aggregateTableName);
             builder.HasKey(c => c.Id);
 
-            builder.Property(c => c.Description);
-            builder.Property(c => c.Name);
             builder.Property(c => c.OwnerId);
             builder.Property(c => c.SenderProvidedChargeId);
             builder.Property(c => c.Resolution);
             builder.Property(c => c.Type);
-            builder.Property(c => c.TaxIndicator);
-            builder.Property(c => c.TransparentInvoicing);
-            builder.Property(c => c.VatClassification);
-            builder.Property(c => c.StartDateTime);
-            builder.Property(c => c.EndDateTime);
-
             builder.OwnsMany(c => c.Points, ConfigurePoints);
+            builder.OwnsMany(c => c.Periods, ConfigurePeriods);
 
             // Enable EF Core to hydrate the points
             builder.Metadata
                 .FindNavigation(nameof(Charge.Points))
                 .SetPropertyAccessMode(PropertyAccessMode.Field);
+
+            // Enable EF Core to hydrate the periods
+            builder.Metadata
+                .FindNavigation(nameof(Charge.Periods))
+                .SetPropertyAccessMode(PropertyAccessMode.Field);
+        }
+
+        private void ConfigurePeriods(OwnedNavigationBuilder<Charge, ChargePeriod> periods)
+        {
+            // This field is defined in the SQL model (as a foreign key)
+            periods.WithOwner().HasForeignKey($"{_aggregateTableName}Id");
+
+            var tableName = $"{nameof(ChargePeriod)}";
+            periods.ToTable(tableName);
+
+            // This is a database-only column - doesn't exist in domain model as period is not an aggregate
+            periods.Property<Guid>("Id").ValueGeneratedOnAdd();
```

Comment:
`Id` is already set by `ChargeFactory`:
```
var period = new ChargePeriod(
                Guid.NewGuid(),
```
so we should not ask the database to set it, too.

```suggestion
            periods.Property(x => x.Id).ValueGeneratedNever();
```